### DISCUSSION TITLE
Addon naming schema

### DISCUSSION THREAD

**Discussion by krotz-dieter** [Siemens-Healthineers/K2s#335] [General]
With issue
[Addon name schema](https://github.com/Siemens-Healthineers/K2s/issues/334)
I propose some kind of naming schema for addons in order to recognize easier the intention of the addon.

Please  give feedback.

**Comment by jgriegershs** [comment]
> With issue [Addon name schema](https://github.com/Siemens-Healthineers/K2s/issues/334) I propose some kind of naming schema for addons in order to recognize easier the intention of the addon.
> 
> Please give feedback.

I am not sure what *implementation* in *USECASE-IMPLEMENTATION* means. *ai-text* and *ai-image* (domain? kind?) have certainly a different meaning than *ingress-nginx* and *ingress-traefik* (OTS).

Maybe it makes sense to offer one *ingress* addon with *nginx*, *traefik*, etc. as options/parameters since we already have addons with this semantic, e.g. *dashboard*, *monitoring* or *registry*. This makes it also easier if we have to replace or remove an OTS (see *Grafana* replacement), because removing a parameter option is less effort than removing a complete addon (even though removing a parameter might still be a breaking change).

Based on that we would have something like that:
- dashboard (nginx | traefik | none)
- ingress (nginx | traefik)
- gateway (nginx)
- exthttpaccess
- logging (fluentbit + opensearch)
- metrics-server
- monitoring (nginx | traefik | none)
- kubevirt
- gpu-node
- registry (nginx | traefik)
- storage (smb | ??? | etc.)

**Reply by petermicuch** [reply]
@jgriegershs - in case of registry, I do not see nginx or traefik as an option. IMO one should be default and install unless one of the ingresses is not already installed, in which case it would use already installed one.

On the other hand I would see for exthttpaccess option nginx/traefik, as we can also use traefik on windows host to route http/s traffic to cluster.

**Reply by jgriegershs** [reply]
@petermicuch If you run `k2s addons enable registry -h`, you will get the documentation for choosing the ingress controller:

```
Flags:
  -i, --ingress string        Ingress controller to use for exposing registry if not already enabled [ingress-nginx|traefik] (default "ingress-nginx")
```

**Reply by jgriegershs** [reply]
To be precise, I did not distinguish between what we already have and might have :-).
